### PR TITLE
Add UTF-8 chars to create_random_column<string_view> benchmark utility

### PR DESCRIPTION
**Pull request by davidwendt**
This updates the `create_random_column<string_view>` benchmark generate utility to support multi-byte UTF-8 characters. The original code only created columns with ASCII characters. The update also adds the space character which will be useful for text-based benchmarks in the future. Only 10 UTF-8 characters are included so a default distribution will still be mostly ASCII strings.

This will help in providing accurate measurements for adding benchmarks for strings APIs #5698.

This change also includes renaming `DURATION_TO_STRING_BENCH_SRC` to just `STRINGS_BENCH` since this benchmark will be folded into a general strings benchmark executable.

### CODE REVIEW

**Review by vuule — CHANGES_REQUESTED** [review]
Good change. One perf-related suggestion:

**Comment by vuule** [comment]
rerun tests

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 1]
```diff
@@ -347,14 +347,24 @@ void copy_string(cudf::size_type src_idx,
...
-  if (!valid) { cudf::clear_bit_unsafe(column_data.null_mask.data(), idx); }
+  if (!valid) {
+    auto const idx = column_data.offsets.size() - 1;
+    cudf::clear_bit_unsafe(column_data.null_mask.data(), idx);
+    // duplicate the offset value to indicate an empty row
+    column_data.offsets.push_back(column_data.offsets.back());
+    return;
+  }
+  std::vector<char> chars;
+  std::generate_n(std::back_inserter(chars), length, [&]() mutable {
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/benchmarks/common/generate_benchmark_input.cpp:358]: > I think we append to `chars` here to have the string length in bytes available, is that right?

Yes

> I would prefer to append directly to `column_data.chars` and keep track of `column_data.chars.size()` instead to get the length for the offsets.

Ok

**Review by kkraus14 — APPROVED** [review]
cmake lgtm

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 2]
```diff
@@ -347,14 +347,22 @@ void copy_string(cudf::size_type src_idx,
...
+  }
+  std::generate_n(std::back_inserter(column_data.chars), length, [&]() mutable {
+    auto ch = char_gen();
+    if (ch < '\x7F') return static_cast<char>(ch);
+    // x7F is at the top edge of ASCII;
+    // the next set of characters are assigned two bytes
+    column_data.chars.push_back('\xC4');
+    return static_cast<char>(ch + 1);
+  });
+  column_data.offsets.push_back(column_data.chars.size());
```
Comment (vuule) [inline cpp/benchmarks/common/generate_benchmark_input.cpp:365]: omg, so much simpler and cleaner :D

**Review by vuule — APPROVED** [review]

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 3]
```diff
@@ -347,14 +347,22 @@ void copy_string(cudf::size_type src_idx,
...
-  if (!valid) { cudf::clear_bit_unsafe(column_data.null_mask.data(), idx); }
+  if (!valid) {
+    auto const idx = column_data.offsets.size() - 1;
+    cudf::clear_bit_unsafe(column_data.null_mask.data(), idx);
+    // duplicate the offset value to indicate an empty row
+    column_data.offsets.push_back(column_data.offsets.back());
+    return;
+  }
+  std::generate_n(std::back_inserter(column_data.chars), length, [&]() mutable {
+    auto ch = char_gen();
```
Comment (codereport) [inline cpp/benchmarks/common/generate_benchmark_input.cpp:358]:
```suggestion
    auto const ch = char_gen();
```

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 4]
```diff
@@ -371,7 +379,8 @@ std::unique_ptr<cudf::column> create_random_column<cudf::string_view>(data_profi
                                                                       std::mt19937& engine,
                                                                       cudf::size_type num_rows)
 {
-  auto char_dist = [&engine, dist = std::uniform_int_distribution<char>{'!', '~'}]() mutable {
+  auto char_dist = [&engine,  // range 32-127 is ASCII; 127-136 will be multi-byte UTF-8
```
Comment (codereport) [inline cpp/benchmarks/common/generate_benchmark_input.cpp:382]:
```suggestion
  auto const char_dist = [&engine,  // range 32-127 is ASCII; 127-136 will be multi-byte UTF-8
```

File: cpp/benchmarks/common/generate_benchmark_input.cpp [hunk 5]
```diff
@@ -347,14 +347,22 @@ void copy_string(cudf::size_type src_idx,
...
+    // duplicate the offset value to indicate an empty row
+    column_data.offsets.push_back(column_data.offsets.back());
+    return;
+  }
+  std::generate_n(std::back_inserter(column_data.chars), length, [&]() mutable {
+    auto ch = char_gen();
+    if (ch < '\x7F') return static_cast<char>(ch);
+    // x7F is at the top edge of ASCII;
+    // the next set of characters are assigned two bytes
+    column_data.chars.push_back('\xC4');
```
Comment (codereport) [inline cpp/benchmarks/common/generate_benchmark_input.cpp:362]: See comment from dependent PR: https://github.com/rapidsai/cudf/pull/7316#discussion_r570571169

**Review by codereport — CHANGES_REQUESTED** [review]
Couple small changes / question about `generate`

File: cpp/benchmarks/common/generate_benchmark_input.cpp (hunk 4 above)
Comment (davidwendt) [reply cpp/benchmarks/common/generate_benchmark_input.cpp:382]: Looks like `char_dist` cannot be const.

File: cpp/benchmarks/common/generate_benchmark_input.cpp (hunk 5 above)
Comment (davidwendt) [reply cpp/benchmarks/common/generate_benchmark_input.cpp]: Consensus from C++ meetup was to change this to a for-loop.
https://github.com/rapidsai/cudf/pull/7316#discussion_r570698106

**Review by codereport — APPROVED** [review]

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**